src/v/wasm/wasi.h

```diff
@@ -26,31 +43,183 @@
 
 namespace wasm::wasi {
 
+/**
+ * This is the function that the wasi standard expects to be exported to call
+ * into the user's main function.
+ */
 constexpr std::string_view preview_1_start_function_name = "_start";
 
+// Identifiers for errors.
 using errno_t = named_type<uint16_t, struct errc_tag>;
-// https://github.com/WebAssembly/wasi-libc/blob/a6f871343313220b76009827ed0153586361c0d5/libc-bottom-half/headers/public/wasi/api.h#L110-L113C1
 constexpr errno_t ERRNO_SUCCESS = errno_t(0);
-
-// https://github.com/WebAssembly/wasi-libc/blob/a6f871343313220b76009827ed0153586361c0d5/libc-bottom-half/headers/public/wasi/api.h#L250-L253C1
 constexpr errno_t ERRNO_INVAL = errno_t(16);
-
-// https://github.com/WebAssembly/wasi-libc/blob/a6f871343313220b76009827ed0153586361c0d5/libc-bottom-half/headers/public/wasi/api.h#L370-L373
 constexpr errno_t ERRNO_NOSYS = errno_t(52);
-
-// https://github.com/WebAssembly/wasi-libc/blob/a6f871343313220b76009827ed0153586361c0d5/libc-bottom-half/headers/public/wasi/api.h#L370-L373
 constexpr errno_t ERRNO_BADF = errno_t(8);
 
-// https://github.com/WebAssembly/wasi-libc/blob/a6f871343313220b76009827ed0153586361c0d5/libc-bottom-half/headers/public/wasi/api.h#L70C1-L97
+// Identifiers for clocks.
 using clock_id_t = named_type<uint32_t, struct clock_id_tag>;
 constexpr clock_id_t REALTIME_CLOCK_ID = clock_id_t(0);
 constexpr clock_id_t MONOTONIC_CLOCK_ID = clock_id_t(1);
 constexpr clock_id_t PROCESS_CPUTIME_CLOCK_ID = clock_id_t(2);
 constexpr clock_id_t THREAD_CPUTIME_CLOCK_ID = clock_id_t(3);
+
 // A timestamp in nanoseconds
 using timestamp_t = named_type<uint64_t, struct timestamp_tag>;
 // A file descriptor
 using fd_t = named_type<int32_t, struct fd_tag>;
+
+// Type of a subscription to an event or its occurrence.
+using event_type_t = named_type<uint8_t, struct eventtype_tag>;
+/**
+ * The time value of clock `subscription_clock::id` has
+ * reached timestamp `subscription_clock::timeout`.
+ */
+constexpr event_type_t CLOCK_EVENT_TYPE = event_type_t(0);
```

Comment:
Can you clarify here? What isn't uniform?

Reply:
Sorry it's a silly term...just that you can `constexpr event_type_t CLOCK_EVENT_TYPE {0}` if you want. Not important at all.

Reply:
Again see my comment about abseil.io/tips/88, that's my context, but happy to use more of this syntax for initialization.

Reply:
Eh, I thought I'd seen more of it other places in rp, but maybe not. I've looked for specific guidance several times; ironically, that abseil post hasn't appeared in my google results. At any rate I don't have a strong innate preference.

Reply:
Maybe something to decide for the style guide? I don't really care much either but am partial to the advice in the abseil document, most because I think the google style guide is a very reasonable document.

Reply:
Ya, I have seen the google style guide adopted nearly verbatim at a previous workplace (C++ 11/14 shop though, so mileage may vary).